honeybadgerbft/core/honeybadger.py

```diff
@@ -133,13 +144,16 @@ def _send(j, o):
             send_r = _make_send(r)
             recv_r = self._per_round_recv[r].get
             new_tx = self._run_round(r, tx_to_send[0], send_r, recv_r)
-            print('new_tx:', new_tx)
+            for nino in new_tx:
+                logger.debug(f'Node id {self.pid} got from his friends: {nino[:40]}')
 
             # Remove all of the new transactions from the buffer
-            self.transaction_buffer = [_tx for _tx in self.transaction_buffer if _tx not in new_tx]
+            self.transaction_buffer = [_tx for _tx in self.transaction_buffer if _tx not in [t.decode('utf-8') for t in new_tx] and _tx not in tx_to_send]
```

Comment:
Fixed a bug here where new_tx was in bytes but transaction_buffer was a list of str's so the comparison was always false